transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1118,6 +1142,20 @@ private void callHandlerCallbackLater(AbstractChannelHandlerContext ctx, boolean
         }
     }
 
+    private void wakeUpForAutoFlush() {
+        if (autoFlush && ENQUEUE_WAKEUP_TASK_UPDATER.compareAndSet(this, 0, 1)) {
+            if (null == wakeupTask) {
+                wakeupTask = new Runnable() {
+                    @Override
+                    public void run() {
+                        ENQUEUE_WAKEUP_TASK_UPDATER.set(DefaultChannelPipeline.this, 0);
+                    }
+                };
+            }
+            channel().eventLoop().execute(wakeupTask);
+        }
+    }
+
```

Comment:
I think it will not make a big impact actually. But we could do a benchmark to be sure. Would be good to have a "typical" pipeline with like 4 - 8 handlers or so.